src/Microsoft.Health.Fhir.CosmosDb/Features/Operations/Export/CosmosExportOrchestratorJob.cs

```diff
@@ -65,17 +121,17 @@ private static ExportJobRecord CreateExportRecord(ExportJobRecord record, long g
                         record.RequestUri,
                         record.ExportType,
                         format,
-                        record.ResourceType,
+                        string.IsNullOrEmpty(resourceType) ? record.ResourceType : resourceType,
                         record.Filters,
                         record.Hash,
                         record.RollingFileSizeInMB,
                         record.RequestorClaims,
-                        record.Since,
-                        record.Till,
-                        startSurrogateId: null,
-                        endSurrogateId: null,
-                        globalStartSurrogateId: null,
-                        globalEndSurrogateId: null,
+                        since == null ? record.Since : since,
+                        till == null ? record.Till : till,
+                        null,
```

Comment:
Nit: can we keep the named parameters? It helps with readability.

Reply:
Good point - also allows the removal of default values I was passing in. 👍🏽